File: Samples/VB/15-ContextPopup/RibbonContextMenuStrip.vb
Imports System.Text
Imports System.ComponentModel
Imports RibbonLib

Namespace _15_ContextPopup
	Public Class RibbonContextMenuStrip
		Inherits ContextMenuStrip
		Private _contextPopupID As UInteger
		Private _ribbon As Ribbon

		Public Sub New(ByVal ribbon_Renamed As Ribbon, ByVal contextPopupID As UInteger)
			MyBase.New()
			_contextPopupID = contextPopupID
			_ribbon = ribbon_Renamed
		End Sub

		Protected Overrides Sub OnOpening(ByVal e As CancelEventArgs)
			_ribbon.ShowContextPopup(_contextPopupID, Cursor.Position.X, Cursor.Position.Y)
			e.Cancel = True
		End Sub
	End Class
End Namespace

File: Samples/Updated-CS/15-ContextPopup/RibbonContextMenuStrip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.ComponentModel;
using RibbonLib;

namespace _15_ContextPopup
{
    public class RibbonContextMenuStrip : ContextMenuStrip
    {
        private uint _contextPopupID;
        private Ribbon _ribbon;

        public RibbonContextMenuStrip(Ribbon ribbon, uint contextPopupID)
            : base()
        {
            _contextPopupID = contextPopupID;
            _ribbon = ribbon;
        }

        protected override void OnOpening(CancelEventArgs e)
        {
            _ribbon.ShowContextPopup(_contextPopupID, Cursor.Position.X, Cursor.Position.Y);
            e.Cancel = true;
        }
    }
}
